storage/pruning/triePruningStorer.go

```diff
@@ -0,0 +1,205 @@
+
+		if epoch < oldestEpochActive && closeOldPersisters {
+			err = p.Close()
+			if err != nil {
+				log.Debug("persister.Close()", "identifier", args.Identifier, "error", err.Error())
+			}
+		} else {
+			persisters = append(persisters, p)
+			log.Debug("appended a pruning active persister", "epoch", epoch, "identifier", args.Identifier)
+		}
+
+		if !closeOldPersisters {
+			closeOldPersisters = shouldCloseOldPersisters(p)
+		}
+	}
+
+	return persisters, persistersMapByEpoch, nil
+}
+
+func shouldCloseOldPersisters(pd *persisterData) bool {
+	val, err := pd.persister.Get([]byte(common.ActiveDBKey))
+	if err != nil {
+		return false
+	}
+
+	if bytes.Equal(val, []byte(common.ActiveDBVal)) {
+		return true
+	}
+
+	return false
+}
+
+// PutInEpochWithoutCache adds data to persistence medium related to the specified epoch and updates the bloom filter
+func (ps *triePruningStorer) PutInEpochWithoutCache(key []byte, data []byte, epoch uint32) error {
+	ps.lock.RLock()
+	pd, exists := ps.persistersMapByEpoch[epoch]
+	ps.lock.RUnlock()
+	if !exists {
+		return fmt.Errorf("put in epoch: persister for epoch %d not found", epoch)
+	}
+
+	persister, closePersister, err := ps.createAndInitPersisterIfClosedProtected(pd)
+	if err != nil {
+		return err
+	}
+	defer closePersister()
+
+	err = persister.Put(key, data)
+	if err != nil {
+		return err
+	}
+
+	if ps.bloomFilter != nil {
+		ps.bloomFilter.Add(key)
+	}
+
+	return nil
+}
+
+// GetFromOldEpochsWithoutAddingToCache searches the old epochs for the given key without adding to the cache
+func (ps *triePruningStorer) GetFromOldEpochsWithoutAddingToCache(key []byte) ([]byte, error) {
+	v, ok := ps.cacher.Get(key)
+	if ok {
+		return v.([]byte), nil
+	}
+
+	ps.lock.RLock()
+	defer ps.lock.RUnlock()
+
+	if ps.bloomFilter != nil && !ps.bloomFilter.MayContain(key) {
+		return nil, fmt.Errorf("key %s not found in %s", hex.EncodeToString(key), ps.identifier)
+	}
+
+	numClosedDbs := 0
+	for idx := 1; idx < len(ps.activePersisters); idx++ {
+		val, err := ps.activePersisters[idx].persister.Get(key)
+		if err != nil {
+			if err == storage.ErrSerialDBIsClosed {
+				numClosedDbs++
+			}
+
+			continue
+		}
+
+		return val, nil
+	}
+
+	if numClosedDbs+1 == len(ps.activePersisters) && len(ps.activePersisters) > 1 {
+		return nil, storage.ErrSerialDBIsClosed
+	}
+
+	return nil, fmt.Errorf("key %s not found in %s", hex.EncodeToString(key), ps.identifier)
+}
+
+// GetFromLastEpoch searches only the last epoch storer for the given key
+func (ps *triePruningStorer) GetFromLastEpoch(key []byte) ([]byte, error) {
+	ps.lock.RLock()
+	defer ps.lock.RUnlock()
+
+	if ps.bloomFilter != nil && !ps.bloomFilter.MayContain(key) {
+		return nil, fmt.Errorf("key %s not found in %s", hex.EncodeToString(key), ps.identifier)
+	}
+
+	if len(ps.activePersisters) < 2 {
+		return nil, fmt.Errorf("key %s not found in %s", hex.EncodeToString(key), ps.identifier)
+	}
+
+	return ps.activePersisters[1].persister.Get(key)
```

Comment:
hardcoded values `len(ps.activePersisters) < 2` and `ps.activePersisters[1]`
shouldn't we use the num active persisters from config?

Reply:
Refactored, but there is no need for numActivePersisters.